sound/soc/sof/sof-audio.h

```diff
@@ -465,6 +465,57 @@ struct snd_sof_route {
 	void *private;
 };
 
+/**
+ * struct snd_sof_dai_dma_ops - DAI DMA ops
+ * @get_dma_data: Function pointer to get the DAI DMA data
+ * @put_dma_data: Function pointer to release the DAI DMA data
+ * @setup: Function pointer for DAI DMA setup
+ * @reset: Function pointer for DAI DMA reset
+ * @pre_trigger: Function pointer for DAI DMA pre-trigger actions
+ * @trigger: Function pointer for DAI DMA trigger actions
+ * @post_trigger: Function pointer for DAI DMA post-trigger actions
```

Comment:
optional? mandatory?

Reply:
All of our ops in this file are optional unless specified otherwise. I'm not sure we need to add optional to all the ops. It would just make it too verbose.